TLM/TLM/Manager/Impl/LaneArrowManager.cs

```diff
@@ -218,5 +220,230 @@ public class LaneArrowManager
         public static ICustomDataManager<List<Configuration.LaneArrowData>> AsLaneArrowsDM() {
             return Instance;
         }
+
+        /// <summary>
+        /// separates turning lanes for all segments attached to nodeId
+        /// </summary>
+        public void SeparateNode(ushort nodeId) {
+            NetNode node = Singleton<NetManager>.instance.m_nodes.m_buffer[nodeId];
+            if (nodeId == 0)
+                return;
+            if ((node.m_flags & NetNode.Flags.Created) == NetNode.Flags.None)
+                return;
```

Comment:
Here will be style warning, ifs should use { } braces.
Whoever will be merging this should walk through the changes and apply the code style. It is easy to do with Resharper or Rider.

Reply:
I like to make my code more readable! I am going to create a subclass and add curly braces.

> return new value where possible, use out where it gets too complex.

I think I have followed the "use out where it gets too complex" part. In all cases where I use ref and out I either have to because I am using functions that require it or it makes my code more complex if I don't. 

for example here I input 3 values and tweaking them only in case they are zero. returning 3 tweaked values is going to make the code more complex specially considering that this helper is called multiple times in a row.
`AvoidZero3_Helper(ref int x, ref int y, ref int z)
`

This peace of code I copied from LaneConnectionManager.cs:
`            Services.NetService.IterateNodeSegments(
                nodeId,
                (ushort otherSegmentId, ref NetSegment otherSeg) => {
`

I cannot see how can I avoid using this code without making it too complicated:
`if (!lhd) {`
`//if right hand drive then favour the more difficult left turns.`
`if (leftLanesCount == 0)`
`    DistributeLanes2(srcLaneCount, forwardLanesCount, rightLanesCount, out f, out r);`
`else if (rightLanesCount == 0)`
`    DistributeLanes2(srcLaneCount, leftLanesCount, forwardLanesCount, out l, out f);`
`else {`
`    //forwarLanesCount == 0`
`    DistributeLanes2(srcLaneCount, leftLanesCount, rightLanesCount, out l, out r);`
`}`
`}`
`else {`
`//if left hand drive then favour the more difficult right turns.`
`if (leftLanesCount == 0)`
`    DistributeLanes2(srcLaneCount, rightLanesCount, forwardLanesCount, out r, out f);`
`else if (rightLanesCount == 0)`
`    DistributeLanes2(srcLaneCount, forwardLanesCount, leftLanesCount, out f,  out l);`
`else //forwarLanesCount == 0 `
`DistributeLanes2(srcLaneCount, rightLanesCount, leftLanesCount, out r,  out l);`
`}
`


So if you have any specific place in mind that you think I should avoid use ref and out please let me know. :)

Reply:
In general `x = helper(x, ref y, ref z)` is more visual than `helper(ref x, ref y, ref z)` because it shows that x is updated, but so also y and z can be updated because they are ref :) There is no clear benefit of touching that code. I am not sure how to make it look better. I'll wait till you separate the code into its own namespace or a class, and will have a look again.